Mathlib/MeasureTheory/Integral/SobolevInequality.lean

```diff
@@ -0,0 +1,610 @@
+/-
+Copyright (c) 2024 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Heather Macbeth
+-/
+import Mathlib.Analysis.Calculus.Deriv.Pi
+import Mathlib.Analysis.InnerProductSpace.EuclideanDist
+import Mathlib.Analysis.InnerProductSpace.NormPow
+import Mathlib.Data.Finset.Interval
+import Mathlib.MeasureTheory.Integral.IntegralEqImproper
+
+/-!
+# Gagliardo-Nirenberg-Sobolev inequality
+
+In this file we prove the Gagliardo-Nirenberg-Sobolev inequality.
+This states that for compactly supported `C¹`-functions between finite dimensional vector spaces,
+we can bound the `L^p`-norm of `u` by the `L^q` norm of the derivative of `u`.
+The bound is up to a constant that is independent of the function `u`.
+Let `n` be the dimension of the domain.
+
+## Main results
+
+* `MeasureTheory.snorm_le_snorm_fderiv_of_eq`:
+  The bound holds for `1 ≤ p < n` and `q⁻¹ = p⁻¹ - n⁻¹`
+* `MeasureTheory.snorm_le_snorm_fderiv_of_le`:
+  The bound holds when `1 ≤ p < n`, `0 ≤ q` and `p⁻¹ - n⁻¹ ≤ q⁻¹`.
+  Note that in this case the constant depends on the support of `u`.
+-/
```

Comment:
Could you sketch in the file-level docstring the proof you are following, first in informal terms and then describing the formalization concepts you use to avoid DTT?

Reply:
Done, and I also further gave a proof sketch in 5 long proofs.